x/incentive/keeper/genesis.go

```diff
@@ -0,0 +1,165 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/store/prefix"
+	"github.com/babylonlabs-io/babylon/x/incentive/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// InitGenesis performs stateful validations and initializes the keeper state from a provided initial genesis state.
+func (k Keeper) InitGenesis(ctx context.Context, gs types.GenesisState) error {
+	for _, entry := range gs.BtcStakingGauges {
+		// check that height is less than current height
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
```

Comment:
question: does the context have correct block height at `InitGenesis`? Like does it loads from the genstate the last block height it stopped as well? 

If yes, nice check!

Reply:
yes it does! However, with the current binary this will fail anyways due to a check in the `InitEpoch` called in the epoching `InitGenesis`

```
panic: InitEpoch can be invoked only at genesis. Invoked at height 6
```